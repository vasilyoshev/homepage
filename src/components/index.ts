export * from './Header/Header';
export * from './AddWidgetFab/AddWidgetFab';
export * from './ManageWidgetsDialog/ManageWidgetsDialog';
export * from './TodoWidget/AddTodoForm/AddTodoForm';
export * from './TodoWidget/TodoList/TodoList';
export * from './TodoWidget/TodoListItem/TodoListItem';
export * from './CryptoCurrencyTrackerWidgetTable/CryptoCurrencyTable';
export * from './CryptoCurrencyTrackerWidgetTableMobile/CryptoCurrencyTableMobile';
export * from './CryptoCurrencyWidgetSearchBar/CryptoCurrencyWidgetSearchBar';
export * from './TimerWidgetControls/TimerWidgetControls';
export * from  './WeatherWidgetLocationSearch/WeatherWidgetLocationSearch';
export * from './WeatherWidgetLocationTable/WeatherWidgetLocationTable';
export * from './WeatherWidgetAlertMessage/WeatherWidgetAlertMessage';
export * from './WeatherWidgetEntry/WeatherWidgetEntry';
export * from './WeatherWidgetSummary/WeatherWidgetSummary';
