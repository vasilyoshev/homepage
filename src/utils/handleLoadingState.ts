import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
