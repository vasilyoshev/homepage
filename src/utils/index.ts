export * from './handleLoadingState';
export * from './testUtils';
