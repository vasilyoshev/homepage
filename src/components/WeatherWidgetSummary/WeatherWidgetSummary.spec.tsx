import { render } from 'utils';
import { WeatherWidgetSummary } from 'components';

describe('WeatherWidgetSummary', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<WeatherWidgetSummary location={null}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
