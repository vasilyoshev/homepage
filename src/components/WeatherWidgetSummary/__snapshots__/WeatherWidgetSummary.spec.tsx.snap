// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherWidgetSummary renders successfully 1`] = `<DocumentFragment />`;
